File: Team_Gamma/ecommerce-main/lib/models/product.dart
class Product {
  final String id;
  final String name;
  final String description;
  final double price;
  final String imageUrl;
  final double rating;
  final int reviews;
  bool isInCart;

  Product({
    required this.id,
    required this.name,
    required this.description,
    required this.price,
    required this.imageUrl,
    required this.rating,
    required this.reviews,
    this.isInCart = false,
  });

  Product copyWith({
    String? id,
    String? name,
    String? description,
    double? price,
    String? imageUrl,
    double? rating,
    int? reviews,
    bool? isInCart,
  }) {
    return Product(
      id: id ?? this.id,
      name: name ?? this.name,
      description: description ?? this.description,
      price: price ?? this.price,
      imageUrl: imageUrl ?? this.imageUrl,
      rating: rating ?? this.rating,
      reviews: reviews ?? this.reviews,
      isInCart: isInCart ?? this.isInCart,
    );
  }
}

File: your-project/backend/models/Product.js
// import mongoose from "mongoose";

// const productSchema = new mongoose.Schema({
//     name: String,
//     description: String,
//     price: Number,
//     category: String,
//     createdAt: { 
//         type: Date, 
//         default: Date.now 
//     }
// });

// export default mongoose.model("Product", productSchema);


import mongoose from "mongoose";

const productSchema = new mongoose.Schema(
  {
    name: { type: String, required: true },
    description: { type: String, required: true },
    price: { type: Number, required: true },
    category: { type: String, required: true },
    stock: { type: Number, default: 0 }, // Optional stock field
    image: { type: String }, // Store image URL if needed
  },
  { timestamps: true }
);

const Product = mongoose.model("Product", productSchema);
export default Product;
